src/mds/MDSRank.cc

```diff
@@ -2658,16 +2658,29 @@ void MDSRankDispatcher::handle_asok_command(
   } else if (command == "session evict" ||
 	     command == "client evict") {
     std::lock_guard l(mds_lock);
+    bool yes_i_really_mean_it = false;
     std::vector<std::string> filter_args;
-    cmd_getval(cmdmap, "filters", filter_args);
+    cmd_getval(cmdmap, "yes_i_really_mean_it", yes_i_really_mean_it);
+    bool got_val = cmd_getval(cmdmap, "filters", filter_args);
 
     SessionFilter filter;
     r = filter.parse(filter_args, css.get());
     if (r != 0) {
       r = -CEPHFS_EINVAL;
       goto out;
     }
-    evict_clients(filter, on_finish);
+    if (got_val){
+      evict_clients(filter, on_finish);
+    }
+    else{
+      if (!yes_i_really_mean_it){
+        *css << "evicting clients require yes_i_really_mean_it to be passed";
+        goto out;
+      }
+      else{
+        evict_clients(filter, on_finish);
+      }
```

Comment:
if there is no `yes_i_really_mean_it` then it is anyways never going to enter this `else`. You can get rid of this block.

Reply:
this `else` block handles the case when no filters are there, but user passes `--yes_i_really_mean_it`.

Reply:
I understand that but what I mean is if it is:

```
    else{
      if (!yes_i_really_mean_it){
        *css << "evicting clients require yes_i_really_mean_it to be passed";
        goto out;
      }
      evict_clients(filter, on_finish);
    }
```
then it doesn't do any harm and the else inside else is sort of not needed; if there is no `yes_i_really_mean_it` then we are anyways jumping to `out`. This is sort of a nit-picking so its absolutely up to on how you want it.

Reply:
I would restructure this so that there is only one `evict_clients(filter, on_finish);` call.  Having identical calls in different branches is a code smell.

Reply:
Fine!!